Use of `multi return` types in Generics leads to C error

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
### V Doctor:

```sh
OS: linux, Ubuntu 20.04.4 LTS (WSL 2)
Processor: 12 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i5-11400H @ 2.70GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /home/apocryphon/tests/v
vmodules: /home/apocryphon/.vmodules
vroot: /home/apocryphon/packages/v
vexe: /home/apocryphon/packages/v/v
vexe mtime: 2022-08-06 16:46:27
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 d6de533.d6b594c

Git version: git version 2.25.1
Git vroot status: weekly.2022.31-22-gd6b594c4
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452
```

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
### What did you do?

```v nofmt
struct Tuple<T> {
    data T
}

fn main() {
    sample := Tuple<(int, int)>{}
    println(sample)
}
```

### What did you expect to see?

> There is a related discussion about this in #15300. To make it short, the code above (with some additions) should result in a compilation error or a successful program with behavior resembling tuples (like the ones in Python).

Related to:
- #15359
- #15083 
- #15366
- #15300

### What did you see instead?
**`v -cg`:**
```failcompile
/tmp/v_1000/m2.12566730850537185187.tmp.c:1074: error: field 'data' has incomplete type
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```
**`v -cg -prod`:**
```
/tmp/v_1000/m2.16452454173144520647.tmp.c:1075:23: error: field ‘data’ has incomplete type
 1075 |  multi_return_int_int data;
      |                       ^~~~
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/parser/parse_type.v
+++ b/vlib/v/parser/parse_type.v
@@ -675,6 +675,9 @@ pub fn (mut p Parser) parse_generic_inst_type(name string) ast.Type {
 			is_instance = true
 		}
 		gts := p.table.sym(gt)
+		if gts.kind == .multi_return {
+			p.error_with_pos('cannot use multi return as generic concrete type', type_pos)
+		}
 		if !is_instance && gts.name.len > 1 {
 			p.error_with_pos('generic struct parameter name needs to be exactly one char',
 				type_pos)

--- a/vlib/v/parser/tests/generic_struct_type_using_multi_return_err.out
+++ b/vlib/v/parser/tests/generic_struct_type_using_multi_return_err.out
@@ -1,0 +1,7 @@
+vlib/v/parser/tests/generic_struct_type_using_multi_return_err.vv:6:18: error: cannot use multi return as generic concrete type
+    4 |
+    5 | fn main() {
+    6 |     sample := Tuple<(int, int)>{}
+      |                     ~~~~~~~~~~
+    7 |     println(sample)
+    8 | }

--- a/vlib/v/parser/tests/generic_struct_type_using_multi_return_err.vv
+++ b/vlib/v/parser/tests/generic_struct_type_using_multi_return_err.vv
@@ -1,0 +1,8 @@
+struct Tuple<T> {
+	data T
+}
+
+fn main() {
+	sample := Tuple<(int, int)>{}
+	println(sample)
+}

